src/Std/Internal/Async/DNS.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time
+import Std.Internal.UV
+import Std.Internal.Async.Basic
+import Std.Net.Addr
+
+namespace Std
+namespace Internal
+namespace IO
+namespace Async
+namespace DNS
+
+open Std.Net
+
+/--
+This type specifies the desired address family for the returned addresses in `getAddrInfo`.
+-/
+inductive AddressFamily
+
+  /--
+  IPv4 address family.
+  -/
+  | ipv4
+
+  /--
+  IPv6 address family.
+  -/
+  | ipv6
+deriving Repr, DecidableEq
+
+/--
+Describes the type of a socket.
+-/
+inductive SocketType
+
+  /--
+  Stream socket, typically used for TCP connections.
+  -/
+  | stream
+
+  /--
+  Datagram socket (SOCK_DGRAM), typically used for UDP connections.
+  -/
+  | datagram
+
+  /--
+  Raw socket (SOCK_RAW), used for low-level network protocols.
+  -/
+  | raw
+deriving Repr, DecidableEq
+
+/--
+Specifies the transport protocol for the DNS search.
+-/
+inductive Protocol
+
+  /--
+  TCP protocol.
+  -/
+  | tcp
+
+  /--
+  UDP protocol.
+  -/
+  | udp
+
+  /--
+  Raw IP packets.
+  -/
+  | raw
+deriving Repr, DecidableEq
```

Comment:
Given that we do not return a proper `SocketAddress` (as is sensible) we should also not expose this particular part of the DNS API as it is only relevant if we expose `SocketAddress`es directly